### PR TITLE
Introduce assert_gui_app

### PR DESCRIPTION
**Pull request by DimStar77**
Most simple function to test a gui application in one command.

optionally installs, starts, asserts screen and closes app again

### CODE REVIEW

File: lib/utils.pm [hunk 1]
```diff
@@ -87,6 +88,26 @@ sub prepare_system_reboot {
...
+#                         mask the fact that the app is not on the media
+#   exec_param: string => When calling the application, pass this parameter on the command line
+#   remain: boolean    => If set to true, do not close the application when tested it is
+#                         running. This can be used if the application shall be tested further
+
+sub assert_gui_app {
+	my ($application, %args) = @_;
+	ensure_package_installed($application) if $args{install};
+	x11_start_program("$application $args{exec_param}");
+	assert_screen("test-$application-1");
```
Comment (coolo) [inline lib/utils.pm]: I assume you want to save the eog tag, but please not these generated tags. Use something more creative and retag the needles.

Comment (DimStar77) [inline lib/utils.pm]: ok; I initially had "test-$application-started; which would be more in line what the needle checks for.

acceptable?

Comment (okurz) [inline lib/utils.pm]: +1

File: lib/utils.pm [hunk 2]
```diff
@@ -87,6 +88,26 @@ sub prepare_system_reboot {
...
+#   exec_param: string => When calling the application, pass this parameter on the command line
+#   remain: boolean    => If set to true, do not close the application when tested it is
+#                         running. This can be used if the application shall be tested further
+
+sub assert_gui_app {
+    my ($application, %args) = @_;
+    ensure_package_installed($application) if $args{install};
+    x11_start_program("$application $args{exec_param}");
+    assert_screen("test-$application-started");
+    send_key "alt-f4" unless $args{remain};
```
Comment (okurz) [inline lib/utils.pm]: As it is not ensured the application already got the command before exiting `assert_gui_app` it should be made sure the window closed. If we are lucky `wait_screen_change { send_key 'alt-f4' } unless $args{remain};` might be good enough

Comment (DimStar77) [inline lib/utils.pm]: as you run this from within a x11base app test, you get the standard 'wait for the desktop to show; before the test terminates

Comment (okurz) [inline lib/utils.pm]: yes, but only if the test writer does not calling something after `assert_gui_app`

Comment (DimStar77) [inline lib/utils.pm]: sounds a bit theoretical - if I want to continue testing this app post the minimal test then I'd give remain=1 to the function. Performing more testing after closing the app would sound a bit strange.

But if there is agreement to use ```wait_screen_change { send_key 'alt-f4' } unless $args{remain};``` then I'm fine changing this

**Pull request merged**